src/ess/imaging/io.py

```diff
@@ -93,8 +97,62 @@ def load_nexus_histogram_mode_detector(
     return HistogramModeDetector(dg)
 
 
-def separate_detector_images(dg: HistogramModeDetector) -> HistogramModeDetectorData:
-    return HistogramModeDetectorData(sc.sort(dg['data'], 'time'))
+Dim1AxisCoord = NewType("Dim1AxisCoord", sc.Variable | None)
+"""Dim1 axis coordinate."""
+Dim2AxisCoord = NewType("Dim2AxisCoord", sc.Variable | None)
+"""Dim2 axis coordinate."""
+
+
+MinDim1 = NewType("MinDim1", sc.Variable | None)
+"""Minimum value of the first dimension."""
+MaxDim1 = NewType("MaxDim1", sc.Variable | None)
+"""Maximum value of the first dimension."""
+MinDim2 = NewType("MinDim2", sc.Variable | None)
+"""Minimum value of the second dimension."""
+MaxDim2 = NewType("MaxDim2", sc.Variable | None)
+
+
+def _assign_coord_or_make_one(
+    da: sc.DataArray, coord_name: str, coord: sc.Variable | None
+) -> None:
+    da.coords[coord_name] = coord or sc.arange(
+        dim=coord_name, start=0, stop=da.sizes[coord_name]
+    )
+
+
+def _crop_da_by_coord(
+    da: sc.DataArray,
+    dim: str,
+    min_coord: sc.Variable | None,
+    max_coord: sc.Variable | None,
+) -> sc.DataArray:
+    if min_coord is not None and max_coord is not None:
+        return da[dim, min_coord:max_coord]
+    elif min_coord is not None and max_coord is None:
+        return da[dim, min_coord:]
+    elif min_coord is None and max_coord is not None:
+        return da[dim, :max_coord]
+    else:
+        return da
```

Comment:
```suggestion
    return da[dim, min_coord:max_coord]
```
You are reimplementing the behaviour of `slice` here.

Reply:
Ah... so I don't have to handle `None`...? That's great.

Reply:
I fixed it!